stellarobservatory/quorum_lachowski_test.py

```diff
@@ -0,0 +1,73 @@
+"""Test for Lachowski's quorum intersection function"""
+from .quorum_lachowski import contract_to_maximal_quorum, has_quorum_intersection
+
+def test_contract_to_maximal_quorum():
+    """Test contract_to_maximal_quorum()"""
+    # pylint: disable=duplicate-code
+    quorum_slices_by_public_key = {
+        'A': [{'A', 'B'}, {'A', 'C'}, {'A', 'B', 'C'}],
+        'B': [{'A', 'B'}],
+        'C': [{'A', 'B', 'C', 'D'}]
+    }
+    # Quorum (gets contracted):
+    assert contract_to_maximal_quorum({'A', 'B', 'C'}, quorum_slices_by_public_key) == {'A', 'B'}
+    # Quorum (already maximal):
+    assert contract_to_maximal_quorum({'A', 'B'}, quorum_slices_by_public_key) == {'A', 'B'}
+    # No quorum:
+    assert contract_to_maximal_quorum({'B', 'C'}, quorum_slices_by_public_key) == set()
+    assert contract_to_maximal_quorum({'A'}, quorum_slices_by_public_key) == set()
+    assert contract_to_maximal_quorum({'B'}, quorum_slices_by_public_key) == set()
+    assert contract_to_maximal_quorum({'C'}, quorum_slices_by_public_key) == set()
+
+def test_has_quorum_intersection_false():
+    """Test failing has_quorum_intersection()"""
+    # Basically two disjoint quorums {A, B} and {D} (two SCCs)
+    quorum_slices_by_node = {
+        'A': [{'A', 'B'}, {'A', 'C'}, {'A', 'B', 'C'}],
+        'B': [{'A', 'B'}],
+        'C': [{'A', 'B', 'C', 'D'}],
+        'D': [{'D'}]
+    }
+
```

Comment:
NIt: for the sake of consistency we could remove this line (the only test where there is a new-line between the case and the assert).
```suggestion
```